todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailPresenter.java

```diff
@@ -81,27 +79,14 @@ private void openTask() {
         }
 
         mTaskDetailView.setLoadingIndicator(true);
-        Subscription subscription = mTasksRepository
+        mSubscriptions.add(mTasksRepository
                 .getTask(mTaskId)
                 .subscribeOn(mSchedulerProvider.computation())
                 .observeOn(mSchedulerProvider.ui())
-                .subscribe(new Observer<Task>() {
-                    @Override
-                    public void onCompleted() {
-                        mTaskDetailView.setLoadingIndicator(false);
-                    }
-
-                    @Override
-                    public void onError(Throwable e) {
-
-                    }
-
-                    @Override
-                    public void onNext(Task task) {
-                        showTask(task);
-                    }
-                });
-        mSubscriptions.add(subscription);
+                .subscribe(this::showTask,
+                        throwable -> {
```

Comment:
It's a super common pattern but since this is educational, I would add a comment per line. // onCompleted // onError // onNext.